tests/dos_files_tests.cpp

```diff
@@ -58,7 +58,7 @@ void assert_DTAExtendName(std::string input,
                           std::string expected_name,
                           std::string expected_ext)
 {
-	char *const input_str = const_cast<char *const>(&input.c_str()[0]);
+	char *const input_str = const_cast<char *>(&input.c_str()[0]);
```

Comment:
If writable access is needed to the underlying char array, the `.data()` member provides a char *, and can be written up to the original length (and is guaranteed contiguous).